### ISSUE TITLE
"g_ck" is not defined warning

### ISSUE THREAD

**Issue by bmoldoveanu** [bhollifield/ServiceNow---OneSearch#2]
Has anyone seen this warning message in relation to OneSearch version 2.18? 
org.mozilla.javascript.EcmaError: "g_ck" is not defined.
Caused by error in <refname> at line 1

==> 1: g_ck

Thank you

**Comment by bhollifield** [comment]
Yes, this is something we see on instances with certain varieties of SSO configured. It is harmless, but I recommend wrapping the g_ck statement in a try/catch if you want to get rid of the console error.

**Comment by bmoldoveanu** [comment]
Specifically what line(s) from the code should be wrapped in try/catch statement? I can't seem to find it. Thanks.

**Comment by bhollifield** [comment]
I believe the only reference is line 6 of the onesearch_js UI Script.

**Comment by bmoldoveanu** [comment]
I've done that and message still shows up. I guess that would suggest the message is not triggered by OneSearch?

**Comment by bhollifield** [comment]
Hmm - Possible, but I've definitely seen this in OneSearch, so I'd expect it's the culprit. Did you do a hard-refresh of the browser and/or cache clear? It may still be using the cached code in the browser. Also, I did see a couple of additional references to g_ck in the onesearch_widget macro, but I really don't think you should need to touch those.

**Comment by bmoldoveanu** [comment]
We replaced line 60 in the UI Macro "onesearch_widget" and we don't see the error message anymore. Possibly because it's been defined as a Phase 1 jelly variable. Do you thin think it might have been a bug?
From 
60		  window.g_ck = "$[g_ck]";
to
60		  window.g_ck = "${g_ck}";

**Comment by bhollifield** [comment]
Thanks for sniffing that out! I've only seen it in certain configurations, and always involving SSO thus far, so probably not a global bug, but interesting to know. I'm sure this will help someone else.

**Comment by bhollifield** [comment]
Hey bmoldoveanu - Just an FYI, the new & supported version of OneSearch is now available via the SN Store. The Share version is necessarily being deprecated, as I'm no longer an employee at SN and thus can no longer updated the version there. The new version comes at a price, which isn't as good as free, but may be worth checking out if you have a long-term strategy around search. Thanks!

https://store.servicenow.com/sn_appstore_store.do#!/store/application/b042b775db473600e0cbda11cf96191b.
